contracts/NFT/JBGenesisNFTDataSource.sol

```diff
@@ -0,0 +1,41 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.6;
+
+import './../interfaces/IJBPayDelegate.sol';
+import './../interfaces/IJBRedemptionDelegate.sol';
+import './../structs/JBPayParamsData.sol';
+import './../structs/JBRedeemParamsData.sol';
+
+contract JBGenesisNFTDataSource is IJBFundingCycleDataSource {
+  IJBPayDelegate public immutable payDelegate;
+
+  constructor(IJBPayDelegate _payDelegate) {
+    payDelegate = _payDelegate;
```

Comment:
since data source is so simple, can just make the same contract both the data source and delegate, and return `self` in payParams. meh?